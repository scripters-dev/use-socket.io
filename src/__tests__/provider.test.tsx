import * as React from 'react';
import { shallow } from 'enzyme';

import Provider from '../provider';
import mockSocket from './mocks/socket-mock';

const url = 'http://local.test/';

jest.mock('socket.io-client', () => () => mockSocket);

describe('Test provider', () => {
    const getWrapper = () => shallow(
        <Provider url={url}>
        Test
        </Provider>,
    );

    it('should match snapshot', () => {
        expect(getWrapper()).toMatchSnapshot();
    });
});
