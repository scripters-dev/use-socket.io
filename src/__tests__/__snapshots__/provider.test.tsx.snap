// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test provider should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "connected": true,
      "disconnect": [Function],
      "emit": [Function],
      "hasListeners": [Function],
      "id": "test",
      "on": [Function],
      "open": [Function],
    }
  }
>
  Test
</ContextProvider>
`;
